client/extensions/woocommerce/app/settings/shipping/shipping-zone-list.js

```diff
@@ -60,7 +60,16 @@ class ShippingZoneList extends Component {
 			);
 		};
 
-		let zonesToRender = loaded ? shippingZones : [ {}, {}, {} ];
+		let zonesToRender;
+
+		if ( loaded ) {
+			zonesToRender = shippingZones;
+		} else if ( shippingZones && shippingZones.length ) {
+			zonesToRender = shippingZones.map( () => ( {} ) );
```

Comment:
The loading state for each row probably occludes the data more than is necessary, but this is a step in the right direction – :+1:!